lib/signaling/v2/twilioconnectiontransport.js

```diff
@@ -0,0 +1,418 @@
+      name: this._name,
+      participant: this._localParticipant.getState(),
+      peer_connections: this._peerConnectionManager.getStates(),
+      type,
+      version: VERSION
+    };
+
+    if (message.type === 'connect') {
+      message.publisher = {
+        name: SDK_NAME,
+        sdk_version: SDK_VERSION,
+        user_agent: this._userAgent
+      };
+      message.token = this._accessToken;
+    } else if (message.type === 'sync') {
+      message.session = this._session;
+      message.token = this._accessToken;
+    } else if (message.type === 'update') {
+      message.session = this._session;
+    }
+
+    const sdpFormat = util.getSdpFormat();
+    if (type === 'connect' && sdpFormat) {
+      message.format = sdpFormat;
+    }
+    this._twilioConnection.sendMessage(message);
+  }
+
+  /**
+   * Disconnect the {@link TwilioConnectionTransport}. Returns true if calling the method resulted
+   * in disconnection.
+   * @param {TwilioError} [error]
+   * @returns {boolean}
+   */
+  disconnect(error) {
+    if (this.state !== 'disconnected') {
+      this.preempt('disconnected', null, [error]);
+      this._sendConnectOrSyncOrDisconnectMessage();
+      this._twilioConnection.close();
+      this._eventPublisher.disconnect();
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Publish an RSP Update. Returns true if calling the method resulted in
+   * publishing (or eventually publishing) the update.
+   * @param {object} update
+   * @returns {boolean}
+   */
+  publish(update) {
+    switch (this.state) {
+      case 'connected':
+        this._twilioConnection.sendMessage(Object.assign({
+          session: this._session,
+          type: 'update',
+          version: VERSION
+        }, update));
+        return true;
+      case 'connecting':
+      case 'syncing':
+        this._updatesToSend.push(update);
+        return true;
+      case 'disconnected':
+      default:
+        return false;
+    }
+  }
+
+  /**
+   * Publish (or queue) an event to the Insights gateway.
+   * @param {string} groupName - Event group name
+   * @param {string} eventName - Event name
+   * @param {object} payload - Event payload
+   * @returns {boolean} true if queued or published, false if disconnected from the Insights gateway
+   */
+  publishEvent(groupName, eventName, payload) {
+    return this._eventPublisher.publish(groupName, eventName, payload);
+  }
+
+  /**
+   * Sync the {@link TwilioConnectionTransport}. Returns true if calling the method resulted in
+   * syncing.
+   * @returns {boolean}
+   */
+  sync() {
+    if (this.state === 'connected') {
+      this.preempt('syncing');
+      this._sendConnectOrSyncOrDisconnectMessage();
+      return true;
+    }
+    return false;
+  }
+}
+
+/**
+ * @event TwilioConnectionTransport#connected
+ * @param {object} initialState
+ */
+
+/**
+ * @event TwilioConnectionTransport#message
+ * @param {object} state
+ */
+
+function reducePeerConnections(peerConnections) {
```

Comment:
I think we should factor these and related methods for reducing RSP message out.

Reply:
@markandrus Since we are not using these functions anywhere else (SIP-based Transport) will go away eventually), I think having them here makes more sense.